pkg/chart/read/ChartReadService.go

```diff
@@ -113,13 +115,14 @@ func (impl *ChartReadServiceImpl) FindLatestChartForAppByAppId(appId int) (chart
 		impl.logger.Errorw("error in fetching app-metrics", "appId", appId, "err", err)
 		return nil, err
 	}
-	chartTemplate, err = impl.chartAdaptor(chart, isAppMetricsEnabled, deploymentConfig)
+	chartTemplate, err = impl.ChartAdaptor(chart, isAppMetricsEnabled, deploymentConfig)
 	return chartTemplate, err
 }
 
-// converts db object to bean
-func (impl *ChartReadServiceImpl) chartAdaptor(chartInput *chartRepoRepository.Chart, isAppMetricsEnabled bool, deploymentConfig *bean2.DeploymentConfig) (*bean.TemplateRequest, error) {
-	if chartInput == nil || chartInput.Id == 0 {
+// ChartAdaptor converts db object chartRepoRepository.Chart to bean.TemplateRequest
+func (impl *ChartReadServiceImpl) ChartAdaptor(chartModel *chartRepoRepository.Chart,
```

Comment:
no need of (impl *chartservice) here, purely an adaptor...better fit outside of readService